src/vmm/src/vstate/kvm.rs

```diff
@@ -82,19 +76,22 @@ impl Kvm {
             let supported_cpuid = kvm_fd
                 .get_supported_cpuid(KVM_MAX_CPUID_ENTRIES)
                 .map_err(KvmError::GetSupportedCpuId)?;
-            let msrs_to_save = crate::arch::x86_64::msr::get_msrs_to_save(&kvm_fd)
-                .map_err(KvmError::GetMsrsToSave)?;
 
             Ok(Kvm {
                 fd: kvm_fd,
                 max_memslots,
                 kvm_cap_modifiers,
                 supported_cpuid,
-                msrs_to_save,
             })
         }
     }
 
+    /// Msrs needed to be saved on snapshot creation.
+    #[cfg(target_arch = "x86_64")]
+    pub fn msrs_to_save(&self) -> Result<MsrList, crate::arch::x86_64::msr::MsrError> {
+        crate::arch::x86_64::msr::get_msrs_to_save(&self.fd)
+    }
+
```

Comment:
Don't have to do it in this PR. But do you think it would make sense to move this arch-specific logic in its own dedicated modules as well? I can do it as a follow up

Reply:
Yeah, I was thinking about that too, but wanted to keep this PR from getting too big. Definitely agree on it being a good idea though

Reply:
ok